kiwi/builder/disk.py

```diff
@@ -855,13 +892,13 @@ def _build_boot_filesystems(
                 'Creating boot(%s) filesystem on %s',
                 boot_filesystem, device_map['boot'].get_device()
             )
-            filesystem = FileSystem.new(
+            with FileSystem.new(
                 boot_filesystem, device_map['boot'], boot_directory
-            )
-            filesystem.create_on_device(
-                label=self.disk_setup.get_boot_label()
-            )
-            system_boot = filesystem
+            ) as filesystem:
+                filesystem.create_on_device(
+                    label=self.disk_setup.get_boot_label()
+                )
+                system_boot = filesystem
         return system_boot, system_efi
```

Comment:
Same here, `system_boot` and `system_efi` are now no longer mounted. This is changed behavior in contrast to previously, where they were mounted until they are garbage collected.

Reply:
correct and a wanted behavior. The former implementation left mounts way too long and several cleanup were racy to fail. This is now much better. I have done many tests to the code hitting Ctrl-C in between of operations which left dead loops or mounts, it's now way better

Reply:
I have switched all kiwi integration tests to main which allows me to build all the images we have with the changed code. So the version of kiwi in Staging is at 10.x.x and I plan to let it run with each iteration of context manager changes and only continue if there are zero failures.

These type of changes are very intrusive to the code. It's different to run context manager based or trust on finalizers. Thus running all build tests with the pull request is imho the only way to do it without too many regressions. I did so with the LoopDevice and that succeeded completely. I would do it with this PR too